root/static/scripts/edit/URLCleanup.js

```diff
@@ -3925,6 +3925,23 @@ const CLEANUPS: CleanupEntries = {
       return {result: false, target: ERROR_TARGETS.URL};
     },
   },
+  'metacritic': {
+    match: [new RegExp('^(https?://)?(www\\.)?metacritic\\.com', 'i')],
+    restrict: [LINK_TYPES.review],
+    clean: function (url) {
+      return url.replace(/^(?:https?:\/\/)?(?:www\.)?metacritic\.com\/music\/([\w-]+\/[\w-]+).*$/, 'https://www.metacritic.com/music/$1');
+    },
+    validate: function (url, id) {
+      switch (id) {
+        case LINK_TYPES.review.release_group:
+          return {
+            result: /^https:\/\/www\.metacritic\.com\/music\/[\w-]+\/[\w-]+$/.test(url),
+            target: ERROR_TARGETS.ENTITY,
+          };
+      }
+      return {result: false, target: ERROR_TARGETS.URL};
+    },
```

Comment:
That seems true, good point - we have a bunch of other uses like this one, like `napster`, which probably need improving then?

Reply:
Thanks for having reordered the tests, it seems to be more logical now, but I’m not too sure that `napster` is incorrect either. For `metacritic` some URL patterns being ignored, whereas `napster` has a better coverage.
In both cases, one of the issues is that the tests are both that the URL is clean and that it matches a specific pattern.  It isn’t worth dwelling on it for now, but it may require a deeper refactoring in the longer term.